force_wfmanager/central_pane/pareto_front.py

```diff
@@ -0,0 +1,82 @@
+from traits.api import (HasStrictTraits, List, Str, Tuple, Instance, Enum,
+                        Property, on_trait_change)
+
+from traitsui.api import View, UItem, Item, VGroup, HGroup
+
+from enable.api import Component, ComponentEditor
+
+from chaco.api import ArrayPlotData, Plot
+
+
+class ParetoFront(HasStrictTraits):
+    #: List of parameter names
+    value_names = List(Str)
+
+    #: List of evaluation steps
+    evaluation_steps = List(Tuple())
+
+    #: List containing the data arrays
+    data_arrays = List(List())
+
+    #: The Pareto Front plot
+    plot = Property(Instance(Component), depends_on=['x', 'y'])
+
+    #: First parameter used for the plot
+    x = Enum(values='value_names')
+
+    #: Second parameter used for the plot
+    y = Enum(values='value_names')
+
+    view = View(VGroup(
+        HGroup(
+            Item('x'),
+            Item('y'),
+        ),
+        UItem('plot', editor=ComponentEditor()),
+    ))
+
+    def _get_plot(self):
+        if self.x is None or self.y is None:
+            return None
+
+        x_index = self.value_names.index(self.x)
+        y_index = self.value_names.index(self.y)
+
+        plot_data = ArrayPlotData()
```

Comment:
try to see if you can add the points without having to recompute the array every time